common/stacks/src/main/scala/weco/api/stacks/services/elastic/ElasticItemLookup.scala

```diff
@@ -64,44 +65,77 @@ class ElasticItemLookup(
     }
   }
 
+  private def buildSourceIdentifierSearchRequest(
+    index: Index
+  )(sourceIdentifier: SourceIdentifier) =
+    search(index)
+      .query(
+        boolQuery
+          .must(termQuery(field = "type", value = "Visible"))
+          .should(
+            termQuery(
+              "data.items.id.sourceIdentifier.value",
+              sourceIdentifier.value
+            )
+          )
+      )
+      // TODO: How many things do we need to retrieve here?
+      .size(10)
+
+  val searchRequestOnIndex = buildSourceIdentifierSearchRequest(index)(_)
+
+  def matchItemsOnWorksBySourceIdentifier(
+    works: Seq[Work[Indexed]],
+    sourceIdentifier: SourceIdentifier
+  ) =
+    works
+      .flatMap { _.data.items }
+      .collect {
+        case item @ Item(id @ IdState.Identified(_, _, _), _, _, _)
+            if id.sourceIdentifier == sourceIdentifier =>
+          // This .asInstanceOf[] is a no-op to help the compiler see what
+          // we can see by reading the code.
+          item.asInstanceOf[Item[IdState.Identified]]
+      }
+
   def bySourceIdentifier(
     sourceIdentifier: SourceIdentifier
   ): Future[Either[ElasticsearchError, Item[IdState.Identified]]] = {
-    // TODO: What if we get something with the right value but wrong type?
-    // We should be able to filter by ontologyType and IdentifierType.
-    val searchRequest =
-      search(index)
-        .query(
-          boolQuery
-            .must(termQuery(field = "type", value = "Visible"))
-            .should(
-              termQuery(
-                "data.items.id.sourceIdentifier.value",
-                sourceIdentifier.value
-              )
-            )
-        )
-        .size(10)
+    val searchRequest = searchRequestOnIndex(sourceIdentifier)
 
     elasticsearchService.findBySearch[Work[Indexed]](searchRequest).map {
       case Left(err) => Left(err)
       case Right(works) =>
-        val item =
-          works
-            .flatMap { _.data.items }
-            .collectFirst {
-              case item @ Item(id @ IdState.Identified(_, _, _), _, _, _)
-                  if id.sourceIdentifier == sourceIdentifier =>
-                // This .asInstanceOf[] is a no-op to help the compiler see what
-                // we can see by reading the code.
-                item.asInstanceOf[Item[IdState.Identified]]
+        matchItemsOnWorksBySourceIdentifier(works, sourceIdentifier).toList match {
+          case item :: _ => Right(item)
+          case List()    => Left(DocumentNotFoundError(sourceIdentifier))
+        }
+    }
+  }
```

Comment:
\*deletes all my comments on this function*

If I checkout your branch locally and do "Find Usages", the only place the `bySourceIdentifier` method is used is in the tests. Let's bin it.